src/python/python.cpp

```diff
@@ -353,6 +353,11 @@ PYBIND11_MODULE(onnxruntime_genai, m) {
       })
       .def("to_token_id", &OgaTokenizer::ToTokenId)
       .def("decode", [](const OgaTokenizer& t, pybind11::array_t<int32_t> tokens) -> std::string { return t.Decode(ToSpan(tokens)).p_; })
+      .def(
+          "apply_chat_template", [](const OgaTokenizer& t, const char* template_str, const char* messages, pybind11::bool_ add_generation_prompt) -> std::string {
+            return t.ApplyChatTemplate(template_str, messages, static_cast<bool>(add_generation_prompt)).p_;
+          },
+          pybind11::arg("template_str") = nullptr, pybind11::arg("messages"), pybind11::arg("add_generation_prompt"))
```

Comment:
Why does it need pybind11::bool_? Is it for the pybind11::arg?
Why do "messages" and "add_generation_prompt" need pybind11::arg params, as they have no defaults set?

Reply:
ah i just figured that was the norm, like using pybind11::array_t<int32_t> instead of std::vector as is done above in other methods, just updated it!

and you're right it's unnecessary i think, fixed!

Reply:
Ah, makes sense. std::vector maps to a python list, vs a numpy array. So that's why that one is explicit (I had that wrong in the tokenizer case). But otherwise we just leave as C++ types when possible.

Reply:
update: "messages" and "add_generation_prompt" need pybind11::arg params as otherwise it fails with: `error: static assertion failed: The number of argument annotations does not match the number of function arguments`

so if you add one optional param, you have to explicitly state the other params too (kind of like in method calls); added them back.